projects/storefrontlib/cms-components/product/product-list/container/product-list-component.service.ts

```diff
@@ -84,10 +85,26 @@ export class ProductListComponentService {
           state.params,
           state.queryParams
         );
-        this.search(criteria);
+        this.searchIfQueryHasChanged(criteria);
       })
     );
 
+  /**
+   * Search only if the previous search query does NOT match the new one.
+   * This prevents repeating product search calls for queries that already have loaded data.
+   */
+  protected searchIfQueryHasChanged(criteria: SearchCriteria) {
+    this.productSearchService
+      .getResults()
+      .pipe(take(1))
+      .subscribe((results) => {
```

Comment:
[QUESTION]
Can it happen that in ngrx will have no "old" state? If yes, then the search will never happen?

Reply:
I have not managed to find a situation where a state will not be returned. In case results are not loaded yet, `getResults()` will immediately return `{}` on subscription.